plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java

```diff
@@ -358,6 +359,45 @@ public void testParquetStatisticsPruning()
         testCountQuery("SELECT count(*) FROM parquet_struct_statistics WHERE row = ROW(2, 'b')", 3, 9);
     }
 
+    @Test
+    public void testPrimitiveFieldsInsideRowColumnPruning()
+    {
+        // Data generated using:
+        // CREATE TABLE nested_fields (id INT, parent STRUCT<child1: BIGINT, child2: STRING>, grandparent STRUCT<parent1 STRUCT<child1: DOUBLE, child2: BOOLEAN>, parent2 STRUCT<child1: DATE>>) USING DELTA LOCATION 's3://krvikash-test/databricks/nested_fields';
+        // INSERT INTO nested_fields VALUES (1, struct('100', 'INDIA'), struct(struct(10.99, true), struct('2023-01-01'))),
+        //    (2, struct('200', 'POLAND'), struct(struct(20.99, false), struct('2023-02-01'))),
+        //    (3, struct('300', 'USA'), struct(struct(30.99, true), struct('2023-03-01'))),
+        //    (4, struct('400', 'AUSTRIA'), struct(struct(40.99, false), struct('2023-04-01'))),
+        //    (5, struct('500', 'JAPAN'), struct(struct(50.99, true), struct('2023-05-01'))),
+        //    (6, struct('600', 'UK'), struct(struct(60.99, false), struct('2023-06-01'))),
+        //    (7, struct('700', 'FRANCE'), struct(struct(70.99, true), struct('2023-07-01'))),
+        //    (8, struct('800', 'BRAZIL'), struct(struct(80.99, false), struct('2023-08-01'))),
+        //    (9, struct('900', 'NAMIBIA'), struct(struct(90.99, true), struct('2023-09-01'))),
+        //    (10, struct('1000', 'RSA'), struct(struct(100.99, false), struct('2023-10-01')));
+
+        assertResultAndSplitCount(
+                "SELECT grandparent.parent1.child1 FROM nested_fields WHERE id > 6",
+                Set.of(90.99, 80.99, 70.99, 100.99),
+                1);
+
+        assertResultAndSplitCount(
+                "SELECT grandparent.parent1.child1 FROM nested_fields WHERE id > 10",
+                Set.of(),
+                0);
+
+        // pruning does not work on primitive fields inside a struct, expected splits should be 1 after file pruning
+        assertResultAndSplitCount(
+                "SELECT grandparent.parent1.child1 FROM nested_fields WHERE parent.child1 > 600",
+                Set.of(90.99, 80.99, 70.99, 100.99),
```

Comment:
nit: Ordering by asc (`70.99, 80.99, 90.99, 100.99` ) will improve the readability.